### PR TITLE
[CM H2]  Fixing small issues

### LINKED ISSUE #1995: CMake add_plugintest corrections
I stumbled upon the following comments (which are spread in at least 2 plugins):
```
# test environment clears env so setting LUA_CPATH is no option
# we workaround this by changing the current directory to our bindings
# output directory + per default lua loads modules from current directory
```

This is no longer true.

### LINKED ISSUE #3665: kdb spec-mount error prints wrong keyname
Points: 1

## Steps to Reproduce the Problem

If `spec:/tests/example` doesn't exist

```sh
kdb meta-get spec:/tests/example mountpoint
#> Key not found

kdb spec-mount spec:/tests/example
```

Alternatively, if only `mountpoint` the metakey doesn't exist:

```sh
kdb meta-get spec:/tests/example mountpoint
#> Metakey not found

kdb spec-mount spec:/tests/example
```

## Expected Result

`kdb` prints an error that the metakey `mountpoint` wasn't found:

```
No metadata "mountpoint" found on key "spec:/tests/example"
```

## Actual Result

`kdb` prints:

```
No metadata "mountpoint" found below spec/tests/example
```

## System Information

- Elektra Version: master

### LINKED ISSUE #3681: Copyright Year out of date
The year in LICENSE.md still says 2018

https://github.com/ElektraInitiative/libelektra/blob/758d002d44764dabc8cdce4635de8285a5f99e00/LICENSE.md#L1-L3

There may also be other locations that are outdated.

### LINKED ISSUE #3685: remove keyCompareByOwner and keyCompareByNameOwner
in file src/libs/elektra/keyset.c

keyCompareByOwner has to be removed, keyCompareByNameOwner has to be removed and calls have to be replaced with keyCompareByName.

### PR DESCRIPTION
**Pull request by a-kraschitzer**
Closes #3685.
Closes #3665.
Closes #1995.
Closes #3681.

## Basics

These points need to be fulfilled for every PR:

- [X] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [X] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [X] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [X] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [X] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by a-kraschitzer** [comment]
I'm not sure where to put information about `kdb spec-mount error prints wrong keyname #3665` in the next release prep.

**Comment by kodebach** [comment]
> I'm not sure where to put information about `kdb spec-mount error prints wrong keyname #3665` in the next release prep.

You can put it into the "Tools" section.

File: doc/images/flamegraph.svg [hunk 1]
```diff
@@ -414,7 +414,7 @@
 <text  x="374.05" y="127.5" ></text>
 </g>
 <g >
-<title>keyCompareByNameOwner (11,692 samples, 0.07%)</title><rect x="954.2" y="69" width="0.9" height="15.0" fill="rgb(220,212,51)" rx="2" ry="2" />
+<title>keyCompareByName (11,692 samples, 0.07%)</title><rect x="954.2" y="69" width="0.9" height="15.0" fill="rgb(220,212,51)" rx="2" ry="2" />
```
[the remark this answers is not in the document]
Comment (kodebach) [reply doc/images/flamegraph.svg:417]: This is just an SVG that is included in `doc/tutorials/profiling.md`. There is no need to change this file.

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -100,6 +100,10 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 
 - Internal changes to ensure compatibility with the new `elektraNotificationContract`. _(Klemens Böswirth)_
 
+### Lua
+
+- Removed outdated information from docs #1995 _(@a-kraschitzer)_
```
[the remark this answers is not in the document]
Comment (mpranj) [reply doc/news/_preparation_next_release.md:86]: Sorry, I don't understand the question.

In any case: please either add a link to the issue like suggested below or remove the issue number (for all changelog entries). Unfortunately GitHub does not automatically link it.

```suggestion
- Removed outdated information from docs [#1995](https://github.com/ElektraInitiative/libelektra/issues/1995) _(@a-kraschitzer)_
```

File: doc/news/_preparation_next_release.md [hunk 3]
```diff
@@ -99,6 +103,7 @@ The text below summarizes updates to the [C (and C++)-based libraries](https://w
 
 ### Core
 
+- remove keyCompareBy(Name)?Owner #3685 _(@a-kraschitzer)_
```
Comment (mpranj) [inline doc/news/_preparation_next_release.md:106]:
```suggestion
- Remove keyCompareBy(Name)?Owner #3685 _(@a-kraschitzer)_
```

File: src/libs/version_gen.awk [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #
 # SPDX-License-Identifier: BSD-2-Clause-FreeBSD
 #
-# Copyright (C) 2019 Elektra Initiative.
+# copyright BSD License (see LICENSE.md or https://www.libelektra.org)
```
Comment (mpranj) [inline src/libs/version_gen.awk:4]: Capital if it is not the doxygen command `@copyright`.

```suggestion
# Copyright BSD License (see LICENSE.md or https://www.libelektra.org)
```
Applies to all instances.

File: src/tools/kdb/specmount.cpp [hunk 5]
```diff
@@ -83,7 +83,7 @@ void SpecMountCommand::buildBackend (Cmdline const & cl)
 
 	if (!cl.quiet && backends.empty ())
 	{
-		std::cout << "No metadata \"mountpoint\" found below spec" << mp << std::endl;
+		std::cout << "No metadata \"mountpoint\" found on key \"spec" << mp << "\"" << std::endl;
```
Comment (mpranj) [inline src/tools/kdb/specmount.cpp:86]: Does this really print the whole key correctly, or does it just fix the quotes?
According to the issue it should print like this:
```
No metadata "mountpoint" found on key "spec:/tests/example"
```
Note the colon `:` symbol between namespace and the key name.

**Review by mpranj — CHANGES_REQUESTED** [review]
Thank you so much for your contribution!

Looks pretty good but I left some small change requests.

_General note_: In future please stick to our contributing guides. I noticed your commit messages do not adhere to [Elektra's guide](https://github.com/ElektraInitiative/libelektra/blob/master/doc/GIT.md#the-commit-message), but let's ignore it for this PR.

File: src/tools/kdb/specmount.cpp (hunk 5 above)
Comment (a-kraschitzer) [reply src/tools/kdb/specmount.cpp:86]: good catch!

File: src/tools/kdb/specmount.cpp [hunk 6]
```diff
@@ -3,7 +3,7 @@
  *
  * @brief source file of spec mount command
  *
- * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ * @Copyright BSD License (see LICENSE.md or https://www.libelektra.org)
```
Comment (mpranj) [inline src/tools/kdb/specmount.cpp:6]: I suspect doxygen will only accept the command lowercase. Either way, all our doxygen commands are lowercase, so let's stick to it.

```suggestion
 * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
```

File: doc/todo/RELEASE.md [hunk 7]
```diff
@@ -16,20 +16,21 @@ If no release critical problems are found during testing, continue.
 - Run scripts/dev/update-infos-status with arguments for heuristics (--auto) and check
   transitions in plugin status: especially from/to experimental
 - Write release notes (without hash sums, guid or proper download links)
-- cp doc/todo/NEWS.md doc/news/_preparation_next_release.md
+- cp doc/todo/NEWS.md doc/news/\_preparation_next_release.md
```
Comment (mpranj) [inline doc/todo/RELEASE.md:19]: I would have some suggestions for this file, but it's probably too much to leave as comments here.
May I simply push my suggestions to your branch and merge everything in one go?

**Review by mpranj — COMMENTED** [review]
Looks great, I think the PR is done. :+1: 

I noticed some (unrelated) things I would like to improve in the `RELEASE.md`, but it would be too much to leave in comments.
May I simply push my suggestions to your branch and merge everything in one go?

**Comment by mpranj** [comment]
I edited the top post to close the related issues.

File: doc/todo/RELEASE.md (hunk 7 above)
Comment (a-kraschitzer) [reply doc/todo/RELEASE.md:19]: please go ahead! :)

Comment (a-kraschitzer) [reply doc/todo/RELEASE.md:19]: I will leave the style fixes to you then.
https://github.com/ElektraInitiative/libelektra/pull/3776/commits/385bd0cbf587852fcdca3236256d60222f020c5d

**Comment by mpranj** [comment]
Thank you so much!

Merged via #3790, but the commit hashes differ due to a rebase.

**Pull request closed without merging**